datastore_cache_api.py

```diff
@@ -0,0 +1,62 @@
+import datetime
+from google.appengine.ext import ndb
+from cache_api import CacheApi
+import logging
+
+_NO_EXPIRATION_DATETIME = datetime.datetime(year=3000, month=1, day=1)
+
+
+class CacheEntry(ndb.Model):
+    """Datastore model for cache values and expirations."""
+    # The value stored in the cache.
+    value = ndb.PickleProperty()
+    # The datetime when to expire the cache entry, or _NO_EXPIRATION_DATETIME for no expiration.
```

Comment:
Maybe I don't understand caches well but it seems safer to me to set a short default expiration value instead of the default being an unlimited expiration date.

Reply:
I agree that that makes more sense, but this matches the default behavior of the memcache api. Wasn't trying to change behavior in this PR, but can if wanted.

Reply:
don't change behavior